pycarol/data_models/data_models.py

```diff
@@ -554,6 +551,93 @@ def create_mapping(self, staging_name, connector_id=None, connector_name=None, d
+                max_workers: `int` default `None`
+                    Number of workers to use when downloading parquet files with pandas back-end.
+                file_pattern: `str` default `None`
+                    File pattern to filter data when fetching from CDS. e.g.
+                    file_pattern='2019-11-25' will fetch only CDS files that start with `2019-11-25`.
+                return_callback_result `bool` default `False`
+                    If a callback is used, it will return the result of the response of the callback. This will skip all the
+                    operation to merge records and return selected columns.
+                :return:
+                    pd.DataFrame with rejected records.
+                """
+
+        if backend not in ('dask', 'pandas'):
+            raise ValueError(f"Backend options are 'dask','pandas' {backend} was given")
+
+        if callback and not callable(callback):
+            raise TypeError(f'"{callback}" object is not callable')
+
+        if return_dask_graph and backend != 'dask':
+            warnings.warn('`return_dask_graph` has no use when `backend!=dask`')
+
+        import_type = 'golden_rejected'
+
+        storage = Storage(self.carol)
+        token_carolina = storage.backend.carolina.token
+        storage_space = storage.backend.carolina.get_bucket_name(import_type)
+
+        if backend == 'dask':
+            d = _import_dask(
+                storage=storage, dm_name=dm_name,
+                import_type=import_type,
+                merge_records=False,
+                return_dask_graph=return_dask_graph,
+                columns=None, file_pattern=file_pattern,
+            )
+
+        elif backend == 'pandas':
+            import pandas as pd
+            d = _import_pandas(
+                storage=storage, dm_name=dm_name,
+                import_type=import_type, columns=None,
+                callback=callback, max_hits=max_hits,
+                max_workers=max_workers,
+                token_carolina=token_carolina,
+                storage_space=storage_space, file_pattern=file_pattern
+            )
+            if d is None:
+                warnings.warn("No data to fetch!", UserWarning)
+                _field_types = self._get_name_type_DMs(self.get_by_name(dm_name)['mdmFields'])
+                cols_keys = list(_field_types)
+
+                d = pd.DataFrame(columns=cols_keys)
+                for key, value in _field_types.items():
+                    if isinstance(value, dict):
+                        value = "STRING"  # If nested we receive as a `STR`
+                    d.loc[:, key] = d.loc[:, key].astype(_DATA_MODEL_TYPES_MAPPING.get(value.lower(), str), copy=False)
+                return d
+
+        else:
+            raise ValueError(f'backend should be either "dask" or "pandas" you entered {backend}')
+
+        if (return_callback_result) and (callback is not None):
+            return d
+
+        return d
```

Comment:
`d` will be returned anyway... why the if?

Reply:
my mistake. I had to adapt from another function, in this context we don't need it. 
tks